src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2536,306 +2536,318 @@ mgp_error mgp_graph_get_vertex_by_id(mgp_graph *graph, mgp_vertex_id id, mgp_mem
+      const std::vector<std::string> properties_str = std::visit(
+          [property_ids = label_properties_pair.second](const auto *impl) {
+            std::vector<std::string> property_ids_str;
+            property_ids_str.reserve(property_ids.size());
+            std::transform(property_ids.begin(), property_ids.end(), std::back_inserter(property_ids_str),
+                           [impl](const auto &property_id) { return impl->PropertyToName(property_id); });
+            return property_ids_str;
+          },
+          graph->impl);
 
-    mgp_value *mgp_value_label = nullptr;
-    if (auto err_label = mgp_value_make_string(label_id_str.c_str(), memory, &mgp_value_label);
-        err_label != mgp_error::MGP_ERROR_NO_ERROR) {
-      return err_label;
-    }
-    if (auto err_label_into_list = mgp_list_append_extend(label_properties_mgp_list, mgp_value_label);
-        err_label_into_list != mgp_error::MGP_ERROR_NO_ERROR) {
-      return err_label_into_list;
-    }
+      mgp_list *label_properties_mgp_list = nullptr;
+      if (auto properties_mgp_list_err =
+              mgp_list_make_empty(properties_str.size() + 1, memory, &label_properties_mgp_list);
+          properties_mgp_list_err != mgp_error::MGP_ERROR_NO_ERROR) {
+        throw std::logic_error("Cannot create list");
+      }
 
-    for (const std::string &property_str : properties_str) {
-      mgp_value *property_mgp_value = nullptr;
-      if (auto err_str = mgp_value_make_string(property_str.c_str(), memory, &property_mgp_value);
-          err_str != mgp_error::MGP_ERROR_NO_ERROR) {
-        return err_str;
+      mgp_value *mgp_value_label = nullptr;
+      if (auto err_label = mgp_value_make_string(label_id_str.c_str(), memory, &mgp_value_label);
+          err_label != mgp_error::MGP_ERROR_NO_ERROR) {
+        throw std::logic_error("Cannot create label value");
       }
-      if (auto err_list = mgp_list_append_extend(label_properties_mgp_list, property_mgp_value);
-          err_list != mgp_error::MGP_ERROR_NO_ERROR) {
-        return err_list;
+      if (auto err_label_into_list = mgp_list_append_extend(label_properties_mgp_list, mgp_value_label);
+          err_label_into_list != mgp_error::MGP_ERROR_NO_ERROR) {
+        throw std::logic_error("Cannot append label value");
       }
-    }
-    mgp_value value(label_properties_mgp_list, label_properties_mgp_list->GetMemoryResource());
 
-    if (auto err_list = mgp_list_append_extend(*result, &value); err_list != mgp_error::MGP_ERROR_NO_ERROR) {
-      return err_list;
-    }
-  }
+      mgp_value_destroy(mgp_value_label);
 
-  return mgp_error::MGP_ERROR_NO_ERROR;
+      for (const std::string &property_str : properties_str) {
+        mgp_value *property_mgp_value = nullptr;
+        if (auto err_str = mgp_value_make_string(property_str.c_str(), memory, &property_mgp_value);
+            err_str != mgp_error::MGP_ERROR_NO_ERROR) {
+          throw std::logic_error("Cannot create property value");
```

Comment:
Also, I would be here more descriptive, same in other error messages: Failed to list all unique constraints - cannot create property value object. If you agree, please apply similar changes to other error messages